src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs

```diff
@@ -532,6 +538,27 @@ private static void ValidateEqualityComparer(object obj)
             Assert.Equal(obj.GetType().GetGenericArguments()[0], objType.GetGenericArguments()[0]);
         }
 
+        private static bool HasObjectTypeIntegrity(ISerializable serializable)
+        {
+            return !PlatformDetection.IsFullFramework ||
+                !(serializable is NotFiniteNumberException ||
+                (serializable.GetType().IsGenericType && serializable.GetType().GetGenericTypeDefinition() == typeof(HashSet<>)));
```

Comment:
Why is HashSet being excluded?  I thought the HashSet issue was just in corefx... it was netfx as well? (If so that would explain how it came to be in corefx.)

Reply:
🤦‍♂️